### PR TITLE
add undeploy feature

### PR DESCRIPTION
**Pull request by thefab**
a first proposal about an undeploy feature

please tell me if you want this (if yes, I will add some documentation before merge)

### CODE REVIEW

**Comment by thefab** [comment]
documentation added (after a little refactoring)

IMHO, this branch is ready to merge

**Comment by monklof** [comment]
When do we need this function？Could you list some examples？

**Comment by thefab** [comment]
I have a "non root" application which deploy crontab lines during startup:

like
 
```
# BEGIN OF FOOBAR APP CRONTAB
* * * * * crontabline1
* * * * * crontabline2
* * * * * crontabline3
# END OF FOOBAR APP CRONTAB
```

It deploys these lines into the current user crontab (among other crontab lines)

When we stop the application, it removes theses lines with 
undeploy_between ("# BEGIN OF FOOBAR APP CRONTAB", "# END OF FOOBAR APP CRONTAB")

File: README.md [hunk 1]
```diff
@@ -43,6 +43,20 @@ deploycron(content="* * * * * echo hello > /tmp/hello")
...
 ```
 
+and 
+
+```python
+def undeploycron_between(start_line, stop_line):
+```
+
+> Uninstall crontab parts between two lines (included).
+> If the start_line or the stop_line is not found into the installed crontab,
```
Comment (monklof) [inline README.md:53]: suggest adding some introductions to the arguments, for example:
- start_line: the crontab *string* used to determine where to start from.
- stop_line: the crontab *string* used to determine where to stop at.

File: deploycron/__init__.py [hunk 2]
```diff
@@ -42,7 +38,50 @@ def deploycron(filename="", content="", override=False):
...
+    stop_line - stop line to delimit the crontab block to remove
+    """
+    lines_installed = [x.strip() for x in _get_installed_content().splitlines()]
+    start_line = start_line.strip()
+    stop_line = stop_line.strip()
+    if start_line not in lines_installed:
+        return False
+    if stop_line not in lines_installed:
+        return False
+    between = False
```
Comment (monklof) [inline deploycron/__init__.py:57]: consider the following bug cases？
- there are duplicated start_line/end_line in the installed crontabs
- the position of start_line is behind the end_line

**Comment by monklof** [comment]
Ok, the idea seems good, but:
- suggest adding some introductions to the arguments (see the comment)
- and consider the bugs mentioned in the comment

I will merge the request later

thanks a lot and best wishes.

File: README.md (hunk 1 above)
Comment (thefab) [reply README.md:53]: @hawker65: can you have a look at this ? (thanks)

File: deploycron/__init__.py (hunk 2 above)
Comment (thefab) [reply deploycron/__init__.py:57]: @hawker65: can you have a look at this ? (thanks)

**Comment by thefab** [comment]
We close these pull requests to reopen new ones soon

**Pull request closed without merging**